src/nccl_ofi_net.c

```diff
@@ -956,7 +956,13 @@ static ncclResult_t ofi_process_cq(nccl_ofi_t *nccl_ofi_comp)
 		}
 		else if (OFI_UNLIKELY(rc == -FI_EAVAIL)) {
 			rc = fi_cq_readerr(cq, &err_buffer, 0);
-			if (OFI_UNLIKELY(rc < 0)) {
+			if (OFI_UNLIKELY(rc == -FI_EAGAIN)) {
+				/*
+				 * Error not available yet.
+				 * fi_cq_read will keep returning -FI_EAVAIL so just bail out and try again later.
```

Comment:
Have we confirmed that this statement is true? Is there any test for it (maybe in libfabrics)?

Reply:
Yes.  From https://ofiwg.github.io/libfabric/v1.0.0/man/fi_cq.3.html:
> Operations which fail are reported 'out of band'. Such operations are retrieved using the fi_cq_readerr function. When an operation that completes with an unexpected error is inserted into an CQ, it is placed into a temporary error queue. Attempting to read from an CQ while an item is in the error queue results in an FI_EAVAIL failure. Applications may use this return code to determine when to call fi_cq_readerr.

Also, inspecting the Libfabric code confirms this behavior.

Reply:
Thanks for confirming the code. This looks sane :)